src/pages/Admin/Venue/DetailsForm.tsx

```diff
@@ -131,11 +131,12 @@ export const DetailsForm: React.FC<DetailsFormProps> = ({
       if (!user) return;
       try {
         // unfortunately the typing is off for react-hook-forms.
-        if (!!venueId) await updateVenue(vals as VenueInput, user);
+        if (!!venueId)
+          await updateVenue({ ...(vals as VenueInput), id: venueId }, user);
         else await createVenue(vals as VenueInput, user);
```

Comment:
Probably would have been better to use `{` + `}` on the if/else blocks since it flows over more than 1 line. We can use the `withId` helper to simplify this line.